source/normalize-arguments.js

```diff
@@ -165,10 +165,10 @@ module.exports = (url, options, defaults) => {
 
 			if (Reflect.has(error, 'headers') && Reflect.has(error.headers, 'retry-after') && retryAfterStatusCodes.has(error.statusCode)) {
 				let after = Number(error.headers['retry-after']);
-				if (is.number(after)) {
-					after *= 1000;
+				if (is.nan(after)) {
+					after = Date.parse(error.headers['retry-after']) - Date.now();
```

Comment:
This fixes a bug: at first I thought `is.number` won't return `true` for NaN, but I was wrong.

`Math.max` is not needed BTW.